lib/cfn-nag/custom_rules/SecurityGroupEgressPortRangeRule2.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+require 'cfn-nag/violation'
+require_relative 'base'
+
+class SecurityGroupEgressPortRangeRule2 < BaseRule
+  def rule_text
+    'Security Groups with an IpProtocol of -1 found egress with port range'
+  end
+
+  def rule_type
+    Violation::WARNING
+  end
+
+  def rule_id
+    'W40'
+  end
+
+  ##
+  # This will behave slightly different than the legacy jq based rule which was
+  # targeted against inline ingress only
+  def audit_impl(cfn_model)
+    violating_security_groups = cfn_model.security_groups.select do |security_group|
+      violating_egresses = security_group.egresses.select do |egress|
+        egress.ipProtocol == -1 && egress.fromPort != egress.toPort
```

Comment:
so i'm trying to grok this version of the rule vs. the existing rule.  the existing rule warns on any port ranges.... this warns when a rule has all protocols and a port range, so it's going to double up the violations.

i guess the question to ask is whether it should always be a warning where all protocols are allowed?  my gut at this point is to make this rule about just ipProtocol, and leave the port range to the other rule.

on a technical note - i think you need to make sure -1 and "-1" are take care of?

given it's just about the protocol, i''d change the name of the class to reflect that -  SecurityGroupEgressAllProtocolsRule

Reply:
Well, I wasn't happy about the name myself, so I'm glad a more succinct one has been suggested.

Will change code to just regard ipProtocol.

So all port ranges in general are considered suspicious ?

Reply:
yea... "suspicious" might be too strong a word but.... it's a warning to make sure you meant to open up that many ports.  there are legit reasons to do it, so it's not a failing violation....

Reply:
Ah, good.  I knew some clients have to open up the ephemeral port range, because the server would reply on a random port.  But I had no idea how servers behaved.